storagemarket/impl/provider_states.go

```diff
@@ -221,15 +222,21 @@ func (p *Provider) sealing(ctx context.Context, deal MinerDeal) (func(*MinerDeal
 }
 
 func (p *Provider) complete(ctx context.Context, deal MinerDeal) (func(*MinerDeal), error) {
-	// TODO: observe sector lifecycle, status, expiration..
-        err := p.fs.Delete(deal.PiecePath)
-        if err != nil {
-                return nil, err
-        }
+	err := p.fs.Delete(deal.PiecePath)
+	if err != nil {
+		return nil, err
+	}
 	sectorID, offset, length, err := p.spn.LocatePieceForDealWithinSector(ctx, deal.DealID)
 	if err != nil {
 		return nil, err
 	}
+	// TODO: Record actual block locations for all CIDs in piece by improving car writing
+	err = p.pieceStore.AddPieceBlockLocations(deal.Proposal.PieceRef, map[cid.Cid]piecestore.BlockLocation{
+		deal.Ref: {},
```

Comment:
What is needed to store the root CID location here?

Reply:
even for the root, we need to figure out where it sits within the CAR file, so it will be addressed at the same time we address the todo above